### ISSUE TITLE
Re-enable DBSReader_t.py integration tests

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#6068]
Since we're currently using DBS emulator in Jenkins, 7 of those tests failed (thus they were set to integration).
Once we decide what to do, those tests should be enabled again for Jenkins.

**Comment by ticoann** [comment]
@ericvaandering, I just talked to Eric. Eric said in Jenkins test DBSReader_t is still using emulator instead of connecting actual call.

First, how the emulator works.
Emulator switched on and off by class variables.
    following function sets which service should be using emulator. (This should be only set on unittest)
    
    https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/EmulatorSwitch.py#L50

For example it sets the emulator only DBS (So unittest doesn't have to connect dbs)
    https://github.com/dmwm/WMCore/blob/master/test/python/WMCore_t/WMSpec_t/StdSpecs_t/MonteCarlo_t.py#L38

and resets when that unittest is done.
    https://github.com/dmwm/WMCore/blob/master/test/python/WMCore_t/WMSpec_t/StdSpecs_t/MonteCarlo_t.py#L50

So it can be controlled for each test not all the unittest.

Second, I think I know what happens in Jenkins.
I looked at the code, and DBSReader_t.py is not using emulators.  So it should use real DBS not emulator if all the unittest runs in one thread.
Unfortunately, the emulator setting is not thread safe (my fault).  So if Jenkins spawns multiple threads,   emulator set up method has race conditions, since it uses Class method (properties) to set the values.

So I think what happens was depending on when  DBSReader_t.py run, (i.e. while WorkQueue_t.py is running in other thread) It will run the dbs emulator instead of contacting actual dbs. 

We can drop whole emulator setting and use testbed DBS instance for the unittest. Or find some other way to set the emulator and give the fine control to each unittest. However if we keep the emulator we need to maintain those code.
